compiler/AST/iterator.cpp

```diff
@@ -1119,6 +1155,118 @@ buildZip4(IteratorInfo* ii, Vec<BaseAST*>& asts, BlockStmt* singleLoop) {
 }
 
 
+//
+// Handle IBB - Iterator Break Block.
+// An IBB contains the defer actions to be taken when break-ing
+// out of the enclosing loop.
+//
+// createIteratorBreakBlocks() adds an "if" after each "yield"
+// so they look like this:
+//
+//   yield ...;
+//   if gIteratorBreakToken {
+//     return;
+//   }
+//
+// This causes callDestructors to add, before the 'return',
+// the defer actions that are appropriate at this point in the iterator.
+//
+void createIteratorBreakBlocks() {
+  for_alive_in_Vec(CallExpr, yield, gCallExprs)
+    if (yield->isPrimitive(PRIM_YIELD))
```

Comment:
I understand this is quick&dirty but I'd consider it better style to enumerate functions that are iterators & adjust yields within those.